src/pal/pal_posix.h

```diff
@@ -342,6 +349,30 @@ namespace snmalloc
         if (getentropy(&result, sizeof(result)) != 0)
           error("Failed to get system randomness");
         return result;
+#elif defined(SYS_getrandom)
+        union
+        {
+          uint64_t result;
+          char buffer[sizeof(uint64_t)];
+        };
+        int cs;
+        auto current = std::begin(buffer);
+        auto target = std::end(buffer);
+        pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, &cs);
```

Comment:
Why is this required?  I am not familiar with calling syscall directly.

Reply:
this implementation comes from musl (I guess it is used to make sure each getentropy call should not be broken in the middle) but yes, I think we can skip this part.